src/main/java/action/Mark.java

```diff
@@ -0,0 +1,38 @@
+package action;
+import java.io.IOException;
+
+import util.TaskList;
+
+/**
+ * Represents an action to mark a task as completed.
+ * Implements the Action interface.
+ */
+public class Mark implements Action {
+    private final TaskList taskList;
+    private final int i;
```

Comment:
Can consider changing the naming convention for this, even though its only used as indexing.